cmake/FindFS.cmake

```diff
@@ -6,7 +6,11 @@
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR UNIX AND NOT APPLE)
     FIND_LIBRARY(FILESYSTEM_LIBRARY NAME libstdc++fs.a HINTS $ENV{HOME}/local/lib64 $ENV{HOME}/local/lib /usr/local/lib64 /usr/local/lib /opt/local/lib64 /opt/local/lib /usr/lib64 /usr/lib /usr/lib/gcc/*/*/ /usr/local/Cellar/gcc/*/lib/gcc/*/)
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    FIND_LIBRARY(FILESYSTEM_LIBRARY NAME libc++experimental.a HINTS $ENV{HOME}/local/lib64 $ENV{HOME}/local/lib /usr/local/lib64 /usr/local/lib /opt/local/lib64 /opt/local/lib /usr/lib64 /usr/lib /usr/local/Cellar/llvm/*/lib)
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0) # Remove this branch once we drop support for clang/llvm version < 7.0
+        FIND_LIBRARY(FILESYSTEM_LIBRARY NAME libc++experimental.a HINTS $ENV{HOME}/local/lib64 $ENV{HOME}/local/lib /usr/local/lib64 /usr/local/lib /opt/local/lib64 /opt/local/lib /usr/lib64 /usr/lib /usr/local/Cellar/llvm/*/lib)
```

Comment:
`NAMES libc++experimental.a libc++fs.a`?

Reply:
Do you mean `NAMES libc++experimental.a libc++fs.a` instead of the branching? I think this should not work since we still have `libc++experimental.a` even with llvm-7, but the fs stuff got moved to its own library.

Reply:
Then switch the order